### PR TITLE
Redmine#68 - linux-firewall

### PR DESCRIPTION
**Pull request by earl-jann**
* NodeJS API to manipulate firewall rules at run time
  * Modified lohika-spawn to support overloading

### CODE REVIEW

File: node_modules/linux/package.json [hunk 1]
```diff
@@ -1,29 +1,71 @@
...
+  "_inCache": true,
+  "_location": "/linux",
+  "_phantomChildren": {},
+  "_requested": {
+    "raw": "linux/",
+    "scope": null,
+    "escapedName": null,
+    "name": null,
+    "rawSpec": "linux/",
+    "spec": "/home/eatormes/workspace/lohika/node_modules/linux",
```
Comment (joegen) [inline node_modules/linux/package.json]: Why is this pointing to your development workspace?

File: node_modules/linux/package.json [hunk 2]
```diff
@@ -1,29 +1,71 @@
...
+    "type": "directory"
+  },
+  "_requiredBy": [
+    "#USER"
+  ],
+  "_resolved": "file:node_modules/linux",
+  "_shasum": "72b5b940b2b1f3847f0294397272736322ce3813",
+  "_shrinkwrap": null,
+  "_spec": "linux/",
+  "_where": "/home/eatormes/workspace/lohika/node_modules",
```
Comment (joegen) [inline node_modules/linux/package.json]: local workspace?

File: node_modules/linux/sample.js [hunk 3]
```diff
@@ -1,33 +1,197 @@
...
 
-/// sample usage only. note: async run so some *will* fail
+var groupname = 'my_test_group';
 
-linux.userAdd(username, function(err) {
-	if (err){
-		console.log('error in adding user: ' + err);
-	} else {
-		console.log('added user ' + username);
+async.series([
```
Comment (joegen) [inline node_modules/linux/sample.js]: Nice touch, using async module for this.

File: node_modules/linux/src/linux.js [hunk 4]
```diff
@@ -10,4 +10,7 @@ exports.getUsers = require(path.join(__dirname, "getusers.js"));
 exports.groupAdd = require(path.join(__dirname, "groupadd.js"));
 exports.groupDel = require(path.join(__dirname, "groupdel.js"));
 exports.groupMod = require(path.join(__dirname, "groupmod.js"));
-exports.getGroups = require(path.join(__dirname, "getgroups.js"));
\ No newline at end of file
+exports.getGroups = require(path.join(__dirname, "getgroups.js"));
+
+//iptables
+exports.IpTable = require("tesserarius");
```
Comment (joegen) [inline node_modules/linux/src/linux.js]: change this to exports.iptables = require("tesserarius");

File: node_modules/linux/sample.js [hunk 5]
```diff
@@ -1,33 +1,197 @@
...
+]
+ 
+ipTable.set_rules(myChain, rules, (err) => {
+    if(err) {
+        throw err;
+    }
+});
+
+// get chain rules
+ipTable.get_rules(myChain, (err, rules) => {
```
Comment (joegen) [inline node_modules/linux/sample.js]: This does not follow camel case notation for functions.  Please re-export this as iptables.getRules()

File: node_modules/linux/sample.js [hunk 6]
```diff
@@ -1,33 +1,197 @@
...
+    {
+        interface: 'eth0',
+        policy: 'ACCEPT',
+        protocol: 'tcp',
+        destination_port: 22,
+        source: '10.0.12.0/24'
+    }
+]
+ 
+ipTable.set_rules(myChain, rules, (err) => {
```
Comment (joegen) [inline node_modules/linux/sample.js]: camelCase please

File: node_modules/linux/src/linux.js (hunk 4 above)
Comment (joegen) [inline node_modules/linux/src/linux.js]: Nevermind the above comment.  Please re-export the functions one by one instead and change them to camelCase

File: node_modules/linux/package.json (hunk 2 above)
Comment (earl-jann) [inline node_modules/linux/package.json]: hmmm this is an open issue with npm. "npm install" generates these args... https://github.com/npm/npm/issues/10393.

For now, let's just make sure to run https://www.npmjs.com/package/removeNPMAbsolutePaths before committing package.json.

**Pull request closed without merging**